src/main/java/com/checkmarx/flow/service/GitAuthUrlGenerator.java

```diff
@@ -0,0 +1,30 @@
+package com.checkmarx.flow.service;
+
+import com.checkmarx.flow.dto.ScanRequest;
+import com.checkmarx.flow.exception.MachinaRuntimeException;
+import com.checkmarx.sdk.config.Constants;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+@Slf4j
+@Service
+public class GitAuthUrlGenerator {
+
+    public String overrideGitAuthUrlByScmAccessToken(ScanRequest.Repository repoType, String gitUrl, String scmAccessToken) {
+        String gitAuthUrl;
+
+        switch (repoType) {
+            case GITHUB:
+            case ADO:
+                gitAuthUrl = gitUrl.replace(Constants.HTTPS, Constants.HTTPS.concat(scmAccessToken).concat("@"));
+                return gitAuthUrl.replace(Constants.HTTP, Constants.HTTP.concat(scmAccessToken).concat("@"));
+            case BITBUCKET:
+                return gitUrl.replace(Constants.HTTPS, Constants.HTTPS.concat(scmAccessToken).concat("@"));
```

Comment:
Is there an on-prem version of BitBucket? If yes, then gitUrl may start with http and not https.

Reply:
Right... Added also BitBucketServer case.